providers/openfeature-provider-flagd/src/openfeature/contrib/provider/flagd/resolvers/process/file_watcher.py

```diff
@@ -0,0 +1,71 @@
+import json
+import logging
+import os
+import re
+import threading
+import time
+import typing
+
+import yaml
+
+from openfeature.event import ProviderEventDetails
+from openfeature.provider.provider import AbstractProvider
+
+from .flags import Flag
+
+
+class FileWatcherFlagStore:
+    def __init__(self, file_path: str, provider: AbstractProvider):
+        self.file_path = file_path
+        self.provider = provider
+        self.last_modified = 0.0
+        self.flag_data: dict[str, Flag] = {}
+        self.load_data()
+        self.thread = threading.Thread(target=self.refresh_file, daemon=True)
+        self.thread.start()
+
+    def shutdown(self) -> None:
+        pass
+
+    def get_flag(self, key: str) -> typing.Optional[Flag]:
+        return self.flag_data.get(key)
+
+    def refresh_file(self) -> None:
+        while True:
+            time.sleep(1)
```

Comment:
please extract this value to a constant or class field

Reply:
I'd also prefer to have it configurable on init of provider, shouldn't be hard to add

Reply:
Added this as configuration to provider itself which is injected to the file watcher 👍